From 2661a641b53899e994c15195057a8fe3c16becf4 Mon Sep 17 00:00:00 2001
From: Hans Breuer <hans@breuer.org>
Date: Fri, 16 Aug 2013 14:01:26 +0200
Subject: [PATCH 16/24] Bug 672190 - Inputs Methods don't work in integrated
 mode

Connect im-related signals also to the integrated UI canvas
---
 app/interface.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/app/interface.c b/app/interface.c
index dc0bb63..36b9e15 100644
--- a/app/interface.c
+++ b/app/interface.c
@@ -561,6 +561,19 @@ use_integrated_ui_for_display_shell(DDisplay *ddisp, char *title)
                          GDK_POINTER_MOTION_HINT_MASK |
                          GDK_FOCUS_CHANGE_MASK);
 
+  g_signal_connect (GTK_OBJECT (ddisp->container), "focus_out_event",
+		    G_CALLBACK (ddisplay_focus_out_event),
+		      ddisp);
+  g_signal_connect (GTK_OBJECT (ddisp->container), "focus_in_event",
+		    G_CALLBACK (ddisplay_focus_in_event),
+		      ddisp);
+  g_signal_connect (GTK_OBJECT (ddisp->container), "realize",
+		    G_CALLBACK (ddisplay_realize),
+                      ddisp);
+  g_signal_connect (GTK_OBJECT (ddisp->container), "unrealize",
+		    G_CALLBACK (ddisplay_unrealize),
+		      ddisp);
+
   notebook_page_index = gtk_notebook_append_page (GTK_NOTEBOOK(ui.diagram_notebook),
                                                   ddisp->container,
                                                   tab_label_container);
-- 
1.8.4.4

